src/test/scala/io/qbeast/spark/index/SparkPlanAnalyzerTest.scala

```diff
@@ -0,0 +1,99 @@
+package io.qbeast.spark.index
+
+import io.qbeast.QbeastIntegrationTestSpec
+
+class SparkPlanAnalyzerTest extends QbeastIntegrationTestSpec {
+
+  object SparkPlanAnalyzerTesting extends SparkPlanAnalyzer
+
+  private lazy val nonDeterministicUDF = org.apache.spark.sql.functions
+    .udf(() => {
+      scala.util.Random.nextInt()
+    })
+    .asNondeterministic()
+
+  "SparkPlanAnalyzer" should "detect underterminism in LIMIT" in withSparkAndTmpDir {
```

Comment:
"indeterminism"?

Reply:
Let's use a single word for this description: non-deterministic, e.g., non-deterministic query plan, non-deterministic column

Reply:
true